modin/engines/ray/cudf_on_ray/frame/gpu_manager.py

```diff
@@ -14,26 +14,75 @@
 import ray
 import cudf
 import pandas
+from .partition import cuDFOnRayFramePartition ## imported from partition.py.
 
 
-@ray.remote(num_gpus=1)
+# ray remote -- parallelizes a class/function
+@ray.remote(num_gpus=1) 
 class GPUManager(object):
+
+    # constructor/initializer method, takes gpu_id.
     def __init__(self, gpu_id):
         self.key = 0
-        self.cudf_dataframe_dict = {}
+        self.cudf_dataframe_dict = {}  # holds cudf DataFrames or cudf Series.
         self.gpu_id = gpu_id
-
+   
     ## TODO(#45): Merge apply and apply_non_persistent
     def apply_non_persistent(self, first, other, func, **kwargs):
+        """ 
+        Given two keys, apply a function using the values associated with the keys as params.
+        Return the value of the function.
+
+        Parameters
+        ---------
+            first : int
+                The first key. You will get a dataframe out of the dataframe_dict with this, store it into df1.
+            other : int
+                The second key. If it isn't a real key, then it's a none. In such a case, func is called with two params
+                instead of three.
+            func : func
+                A function that we will use/apply on the two other params (first, other).
+            **kwargs: dict
+                An iterable object that corresponds to a dict, if i'm not mistaken.  
+        Returns
+        -------
+            result
+                the result of the function (will be an OID).
+        """
         df1 = self.cudf_dataframe_dict[first]
-        df2 = self.cudf_dataframe_dict[other] if other else None
+        df2 = self.cudf_dataframe_dict[other] if other else None # if other is not passed in, None.
```

Comment:
Apply/Apply_non_persistent should be merged. Ideally both parameters can take either two keys or one key and one OID. The oid is always the other param. I think the comments are fine but most likely will change in the following issue you tackle!